zuul-core/src/test/java/com/netflix/zuul/StaticFilterLoaderTest.java

```diff
@@ -16,45 +16,47 @@
 
 package com.netflix.zuul;
 
-import static org.junit.Assert.assertEquals;
-
+import com.google.common.collect.ImmutableSet;
+import com.google.common.truth.Truth;
 import com.netflix.zuul.filters.FilterType;
 import com.netflix.zuul.filters.ZuulFilter;
 import com.netflix.zuul.filters.http.HttpInboundSyncFilter;
 import com.netflix.zuul.message.http.HttpRequestMessage;
-import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.SortedSet;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
 @RunWith(JUnit4.class)
 public class StaticFilterLoaderTest {
 
-
     private static final FilterFactory factory = new DefaultFilterFactory();
 
     @Test
     public void getFiltersByType() {
-
         StaticFilterLoader filterLoader =
-                new StaticFilterLoader(factory, Arrays.asList(DummyFilter2.class, DummyFilter1.class));
+                new StaticFilterLoader(factory,
+                        ImmutableSet.of(DummyFilter2.class, DummyFilter1.class, DummyFilter22.class));
+
+        SortedSet<ZuulFilter<?, ?>> filters = filterLoader.getFiltersByType(FilterType.INBOUND);
+        Truth.assertThat(filters).hasSize(3);
+        List<ZuulFilter<?, ?>> filterList = new ArrayList<>(filters);
 
-        List<ZuulFilter<?, ?>> filters = filterLoader.getFiltersByType(FilterType.INBOUND);
-        assertEquals(2, filters.size());
-        // Filters are sorted by order
-        assertEquals(DummyFilter1.class, filters.get(0).getClass());
-        assertEquals(DummyFilter2.class, filters.get(1).getClass());
+        Truth.assertThat(filterList.get(0)).isInstanceOf(DummyFilter1.class);
+        Truth.assertThat(filterList.get(1)).isInstanceOf(DummyFilter2.class);
+        Truth.assertThat(filterList.get(2)).isInstanceOf(DummyFilter22.class);
     }
 
     @Test
     public void getFilterByNameAndType() {
         StaticFilterLoader filterLoader =
-                new StaticFilterLoader(factory, Arrays.asList(DummyFilter2.class, DummyFilter1.class));
+                new StaticFilterLoader(factory, ImmutableSet.of(DummyFilter2.class, DummyFilter1.class));
 
         ZuulFilter<?, ?> filter = filterLoader.getFilterByNameAndType("Robin", FilterType.INBOUND);
 
-        assertEquals(DummyFilter2.class, filter.getClass());
+        Truth.assertThat(filter).isInstanceOf(DummyFilter2.class);
```

Comment:
Why use a different assertion style here?

Reply:
It has some very nice list/set/string assertions.   I had added this a few months ago but had to revert it for an unrelated reason.   It's a very nice library to use, and I was planning on replacing some other usages too.